
# HG changeset patch
# User Vladimir Nadvornik <nadvornik@suse.cz>
# Date 1250812164 -7200
# Node ID 016cb2f2fbd035dae812f914fbff17e916fec86a
# Parent c24c1b6787cac93c80689d4fff21fd6930388a20
Fix findBestIcon()

    This patch fixes the issue described here:

    https://bugzilla.novell.com/show_bug.cgi?id=371974

Transplanted from git://repo.or.cz/wmaker-crm.git
commit e113ec10c9add6e3001c583a2c87b4a33cfb255b

--- a/src/wmspec.c	Tue Aug 14 15:32:00 2007 +0400
+++ b/src/wmspec.c	Fri Aug 21 01:49:24 2009 +0200
@@ -402,7 +402,7 @@
      * results in better overall aesthetics -Dan */
     wanted = wPreferences.icon_size * wPreferences.icon_size;
 
-    for (icon=NULL, distance=LONG_MAX, i=0L; i<items-1; ) {
+    for (icon=NULL, distance=wanted, i=0L; i<items-1; ) {
         size = data[i] * data[i+1];
         if (size==0)
             break;

